File: trunk/MyBot/MyBotFSharp/MyBot.fs
namespace RockPaperAzure
open RockPaperScissorsPro

type MyBot() = class
    interface IRockPaperScissorsBot with
        member this.MakeMove(y, o, r) = this.MakeMove(y, o, r)
    end

    // Random sample implementation
    member this.MakeMove(you: IPlayer, opponent: IPlayer, rules: GameRules) = Moves.GetRandomMove()

    // Cycle sample implementation
//    member this.MakeMove(you: IPlayer, opponent: IPlayer, rules: GameRules) = 
//        match (match you.LastMove with
//                | null -> ""
//                | _ -> you.LastMove.ToString().ToLower()) with
//        | "rock" -> Moves.Paper
//        | "paper" -> Moves.Scissors
//        | "scissors" -> if you.HasDynamite then
//                           Moves.Dynamite
//                        else
//                           Moves.WaterBalloon
//        | "dynamite" -> Moves.WaterBalloon
//        | _ -> Moves.Rock

    // BigBang sample implementation
//    member this.MakeMove(you: IPlayer, opponent: IPlayer, rules: GameRules) =
//        match you.NumberOfDecisions with 
//        | five when five < 5 -> Moves.Dynamite
//        | _ -> Moves.GetRandomMove()
end

File: trunk/MyBot/MyBotVB/MyBot.vb
Imports RockPaperScissorsPro

Public Class MyBot
    Implements IRockPaperScissorsBot

    ' Random sample implementation
    Public Function MakeMove(you As IPlayer, opponent As IPlayer, rules As GameRules) As Move Implements IRockPaperScissorsBot.MakeMove
        Return Moves.GetRandomMove()
    End Function

    ' Cycle sample implementation
    'Public Function MakeMove(you As IPlayer, opponent As IPlayer, rules As GameRules) As Move Implements IRockPaperScissorsBot.MakeMove
    '    If you.LastMove = Moves.Rock Then Return Moves.Paper

    '    If you.LastMove = Moves.Paper Then Return Moves.Scissors

    '    If you.LastMove = Moves.Scissors Then
    '        If you.HasDynamite Then
    '            Return Moves.Dynamite
    '        Else
    '            Return Moves.WaterBalloon
    '        End If
    '    End If

    '    If you.LastMove = Moves.Dynamite Then Return Moves.WaterBalloon

    '    Return Moves.Rock
    'End Function

    ' BigBang sample implementation
    'Public Function MakeMove(you As IPlayer, opponent As IPlayer, rules As GameRules) As Move Implements IRockPaperScissorsBot.MakeMove
    '    If you.NumberOfDecisions < 5 Then
    '        Return Moves.Dynamite
    '    Else
    '        Return Moves.GetRandomMove()
    '    End If
    'End Function
End Class